haystack_experimental/components/agents/agent.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import inspect
+from typing import Any, Optional, Union
+
+from haystack import logging
+from haystack.components.agents.agent import Agent as HaystackAgent
+from haystack.components.agents.state.state import _validate_schema
+from haystack.components.agents.state.state_utils import merge_lists
+from haystack.components.generators.chat.types import ChatGenerator
+from haystack.core.component.component import component
+from haystack.core.pipeline.utils import _deepcopy_with_exceptions
+from haystack.dataclasses import ChatMessage
+from haystack.dataclasses.streaming_chunk import StreamingCallbackT
+from haystack.tools import Tool, Toolset
+
+from haystack_experimental.components.tools.tool_invoker import ToolInvoker
```

Comment:
The init of Agent is unchanged, only this import was updated so we use the experimental ToolInvoker